src/lib/phase/end-of-round.js

```diff
@@ -0,0 +1,347 @@
+                }
+            }
+        }
+    }
+    /**
+     * Distributes command tokens to all players.
+     */
+    static distributeCommandTokens() {
+        for (const playerDesk of world.TI4.getAllPlayerDesks()) {
+            const playerSlot = playerDesk.playerSlot;
+            const commandTokenBag =
+                EndStatusPhase.getCommandTokenBag(playerSlot);
+            const count =
+                EndStatusPhase.getNumberOfCommandTokensToDistribute(playerSlot);
+
+            const message = locale("ui.message.distribute_command_tokens", {
+                playerColor: playerDesk.colorName,
+                count: count,
+            });
+            const errorMessage = locale(
+                "ui.message.not_enough_command_tokens",
+                {
+                    playerColor: playerDesk.colorName,
+                }
+            );
+            Broadcast.chatAll(message);
+
+            for (let i = 0; i < count; i++) {
+                if (commandTokenBag.getItems().length === 0) {
+                    Broadcast.chatAll(errorMessage);
+                    break;
+                } else {
+                    const dropPosition = playerDesk.localPositionToWorld(
+                        new Vector(3, 20 + i * 1, 0)
+                    );
+                    commandTokenBag.takeAt(0, dropPosition, true);
+                }
+            }
+        }
+    }
+    /**
+     * Returns all strategy cards to their proper position and rotation.
+     */
+    static returnStrategyCards() {
+        for (const obj of world.getAllObjects()) {
+            if (ObjectNamespace.isStrategyCard(obj)) {
+                const strategyCard = ObjectNamespace.parseStrategyCard(obj);
+                const strategyCardHome = STRATEGY_CARDS.filter((element) =>
+                    element.nsid.includes(strategyCard.card)
+                )[0].pos;
+                const strategyCardRotation = new Rotator(0, -90, 0);
+                obj.setPosition(strategyCardHome, ANIMATION_SPEED);
+                obj.setRotation(strategyCardRotation, ANIMATION_SPEED);
+            }
+        }
+    }
+
+    /**
+     *  If the given card is not faceup, make it face up.
+     * @param {Card} cardObj
+     */
+    static makeFaceUp(cardObj) {
+        assert(cardObj instanceof Card);
+        if (!cardObj.isFaceUp()) {
+            const rotation = cardObj.getRotation();
+            const newRotation = new Rotator(rotation.pitch, rotation.yaw, -180);
+            cardObj.setRotation(newRotation, ANIMATION_SPEED);
+        }
+    }
+
+    /**
+     * Refreshes all planets (that are not on system tiles), technologies, agents
+     * and relics.
+     */
+    static refreshCards() {
+        for (const obj of world.getAllObjects()) {
+            if (obj.getContainer()) {
+                continue; // ignore objects in containers
+            }
+
+            if (ObjectNamespace.isCard(obj)) {
+                if (obj.getStackSize() > 1) {
```

Comment:
CardUtil.isLooseCard has this plus a number of other checks, you might want to use that